### ISSUE TITLE
(libreoffice-still) LibreOffice Still: automatic update broken? update(s) available!

### ISSUE THREAD

**Issue by zWhdmB5T** [chocolatey-community/chocolatey-packages#2347]
### Checklist

- [X] I have checked the moderation queue that no new version has been submitted (only visible when logged in), or the submitted package version has failed the automated checks.

Please navigate to the following [link](https://community.chocolatey.org/packages?q=&moderatorQueue=true&moderationStatus=all-statuses&prerelease=false&sortOrder=package-download-count) to view the moderation queue.

- [X] I have verified that the new version is a stable release.
- [X] I have looked to see if there are any Bug reports that prevents a new version being submitted and that no existing Outdated Reports have been created.
- [X] I have verified the new version has a Windows binary (EXE or MSI installer, Zip or other archive, etc).

### New Software Version

7.5.8.2

### Download location

https://downloadarchive.documentfoundation.org/libreoffice/old/7.5.8.2/

### Package Page

https://github.com/chocolatey-community/chocolatey-packages/tree/master/automatic/libreoffice-streams

**Comment by zWhdmB5T** [comment]
Automatic update of `LibreOffice Still` aka `libreoffice-still` seems to be broken. Currently available at Chocolatey is 7.5.6. But currently available at LO is 7.5.8.2. So, 7.5.7.1 (Bugfix release because of WebP bug), 7.5.8.1 and 7.5.8.2 have been missing. 7.5.8.2 should be the last release of 7.5 cycle: [Release Plan 7.5](https://wiki.documentfoundation.org/ReleasePlan/7.5).

**Comment by zWhdmB5T** [comment]
@choco-bot @ any, anyone? no one for `libreoffice-still`?

**Comment by pauby** [comment]
@zWhdmB5T this is a community maintained repository. The expectation is that you'll get involved, make changes and help to fix, maintain or enhance the package for the benefit of the community.

If you can't / won't and nobody else is willing, then there is no point having the issue open and it will be closed.

**Comment by zWhdmB5T** [comment]
I am willing to, but I am not able to…

(I could try for specific version updater — but I actually don't really know how to fix the automatic updater, sorry.)

(I would spend money for this :-) )

**Comment by zWhdmB5T** [comment]
> @zWhdmB5T this is a community maintained repository. The expectation is that you'll get involved, make changes and help to fix, maintain or enhance the package for the benefit of the community.
> 
> If you can't / won't and nobody else is willing, then there is no point having the issue open and it will be closed.

@pauby Well, I tried… but it seems it needs to get the automatic updater script fixed. But, sorry, I am not able to do this. I am not into programming and coding. All I have my GitHub account for is providing BugReports, Comments and FeautreRequests.

I am also willing to pay for more and better updates. I don't need business/corporate Chocolatey products you offer. I am fine with community/OSS Chocolatey. I just need more and better updates… and I would pay for that. But for this case I don't find any hint on the website… I am asking this need and offering money because whenever I report a package out of date / out of maintenance, you always refer to the community aspect. It's clear, it's the community… And I would help — if I could. All I can do is paying for better service (and I don't need a business/corporate Choco but just more and better updates). Sorry, I really cannot code…

**Comment by pauby** [comment]
> All I can do is paying for better service (and I don't need a business/corporate Choco but just more and better updates)

I know what you meant here with t heabove. But I really just wanted to clarify a couple of points in case others don't.

> paying for better service

There is no _service_ being provided. There are no SLA's, expectations or obligations. The community create and maintain packages in their spare time to help others out. So there is no _service_ to _pay_ for.

When it comes to _better_, again, people do this in their spare time so they do what they can. If you feel you can fix, maintain, update, change or help in any way to make things better then, as part of hte community, that's what we expect. 

> need a business/corporate Choco but just more and better update

Having a _business/corporate Choco_ doesn't change anything in this repository. The repository is maintained by the community who are volunteers who help out in their spare time to provide packages to the Chocolatey Community Repository. I justw ant to stress the word _community_ again. If you have a _business/corporate Choco_ there is no extra service, or anything you pay for on the Chocolatey Community Repository. I, you and anybody with a _business/corporate Choco_ get access to the same packages maintained by the same volunteer maintainers.

> Sorry, I really cannot code…

Neither could I until I learned. I don't want that to be flippant, but I also do want it to come across as we all start and learn from the beginning.

>  and I would pay for that

We don't ask or take any payment for packages

**Comment by zWhdmB5T** [comment]
TL;DR: ticket can be closed (from my side) as I don't have anything serious (like real code) to contribute…

For discussion @pauby

Well, I am not a coder and I actually don't want to become one this life, any more (sorry, but I have my reasons). I just want to use Chocolatey as an ordinary user (like people with much less IT knowledge than me).

My main system is by now a Linux one. There, repository management has been used for quite a long time and is something like state of the art. But it's true: when coming to FlatHub or Snap, it is still much to the broad community to work properly resp. provide updates always on time. Therefore, I was lucky and happy to find Chocolatey for Microsoft Windows, one day back then. I recommend it to family members and friends. I recommend it over WinGet, because WinGet (which is also community run…) has also a lack of updates somewhere/sometimes and it often provides only en-US language (not international or de-DE for my need).

I appreciate and value OpenSourceSoftware and community projects. Not just because they're mostly free of costs (for the user). But for the OpenSource approach (for many different reasons).

What I do is: serving bug reports (sometimes very detailed), asking for feature requests (I try not to be demanding — I am just asking when there's a real need), and I do spend money for donations. Not all projects/maintainers accept donations. But where they do and when I use the software/project quite regularly, I then donate.

I was asking for kind of "premium support" concerning updates in my comment above with the following background:
Chocolatey offers several editions to be paid. I think for a common home user they are not need for technical reason (they offer many features I don't really need). And the Pro Edition does not offer benefit for me for the money I have to pay. But I would be thinking about paying 50 $/y for better/more updates. My rough calculation is as following. If there would be just 2.000 subscriptions for 50 $/y for more/better updates, an employee could be hired caring just for updates during the year.

There are updates that work as automatic updates and such as manual ones. The employee would have to take care for more automatic updates in general (and to care about the automatic updates over time). And he could take care about the manual updates the rest of the time. This would lead to a much better user experience I guess.

Just for discussion, no demanding at all…

**Comment by brogers5** [comment]
The update script is behaving as designed.

As stated in the Notes section of the package's description:

> This package relies on the update service provided by The Document Foundation to determine new updates. As their policy is to distribute to their website immediately and to the update service after 1 to 2 weeks, there will be a delay between a new software update and a new version release of the package.

While it has been a few weeks since v7.5.8's publication on the website (2023-11-02), The Document Foundation's update service is still not offering v7.5.8 as of now, so the update script won't pick up on it for the time being. We'll need to wait out its publication before a package for v7.5.8 can be automatically created.

[Changes for the previous version (v7.5.7)](https://github.com/chocolatey-community/chocolatey-packages/commit/ed17b3b72389197875e634b0b17b53f5d8a1e9c8) were made about a couple months ago, but the resulting package [failed to push to the Community Repository](https://gist.github.com/choco-bot/a14b1e5bfaf70839b338eb1ab7f8226f/e9e3b43d26d61b0b546f2f46eaec03f0b67c34e1#libreoffice-still). A core maintainer may need to force a package rebuild for this version with AU, since the current repository state is up-to-date with what's currently being offered by the update service.

**Comment by zWhdmB5T** [comment]
@brogers5 OK, understood then. Thank you very much for your explanation!

One specific enquiry because of my interest:
Are you talking about:
`chocolatey-packages/automatic/libreoffice-streams/tools/helpers.ps1`
```# Send a generic request to the LibreOffice update service
function GetLatestVersionFromLibOUpdateChecker($userAgent) {
  $url = "https://update.libreoffice.org/check.php"
  $request = [System.Net.WebRequest]::Create($url)
  $request.UserAgent = $userAgent
  $s = $request.GetResponse().GetResponseStream()
  $sr = New-Object -TypeName System.IO.StreamReader($s)
  [xml]$xmlAnswer = $sr.ReadToEnd()
  return $xmlAnswer.description.version
}
```

How does this [request to the LibreOffice update service](https://update.libreoffice.org/check.php) work? In my browser it shows just an error message, but not the version string…

**Comment by pauby** [comment]
> A core maintainer may need to force a package rebuild for this version with AU, since the current repository state is up-to-date with what's currently being offered by the update service.

This was [done](https://github.com/chocolatey-community/chocolatey-packages/commit/8a8f167714803e0a244494e1f6bb227219da3bf2) just now and the missing version was not pushed.

**Comment by brogers5** [comment]
> How does this [request to the LibreOffice update service](https://update.libreoffice.org/check.php) work? In my browser it shows just an error message, but not the version string…

You'd need to [spoof your browser's user agent](https://www.searchenginejournal.com/change-user-agent/368448/) in order for the update service to respond in the expected manner. Check `GetLatestStillVersionFromLibOUpdateChecker` for the value used by the script.

Alternatively, you could check for updates within LibreOffice with a known outdated version.

**Comment by brogers5** [comment]
While looking into this, I've noticed the update service has since started offering v7.5.8 and successfully produced a package for it in my environment. This will likely be resolved with the next scheduled AU run.

**Comment by zWhdmB5T** [comment]
> > How does this [request to the LibreOffice update service](https://update.libreoffice.org/check.php) work? In my browser it shows just an error message, but not the version string…
> 
> You'd need to [spoof your browser's user agent](https://www.searchenginejournal.com/change-user-agent/368448/) in order for the update service to respond in the expected manner. Check `GetLatestStillVersionFromLibOUpdateChecker` for the value used by the script.
> 
> Alternatively, you could check for updates within LibreOffice with a known outdated version.

Thank you!

**Comment by zWhdmB5T** [comment]
> > This was [done](https://github.com/chocolatey-community/chocolatey-packages/commit/8a8f167714803e0a244494e1f6bb227219da3bf2) just now and the missing version was not pushed.
> 
> Looks like [AU didn't even acknowledge the forced build request](https://gist.github.com/choco-bot/a14b1e5bfaf70839b338eb1ab7f8226f/23cd62720ce19efd42a33c7a7b26562e01dedc3e#libreoffice-streams).
> 
> This particular update script is quirky in that `au_GetLatest` will only return a stream if it is created by `AddLibOVersionsToStreams`. `GetLibOVersionsWithoutFromVersion` creates the list of versions that is iterated over. The list starts with the version currently defined in the streams file, but the desired version is deliberately filtered out because "it is the current one that has already been pushed to Chocolatey" which prevents construction of the necessary stream. We probably shouldn't be doing this in a forced update scenario.

Thank you!

**Comment by zWhdmB5T** [comment]
> While looking into this, I've noticed the update service has since started offering v7.5.8 and successfully produced a package for it in my environment. This will likely be resolved with the next scheduled AU run.

Hehe ;-) I just kindly asked the Document Foundation team this afternoon, and they responded soon including this effort.

**Comment by brogers5** [comment]
`libreoffice-still` v7.5.8 has been [produced, pushed](https://gist.github.com/choco-bot/a14b1e5bfaf70839b338eb1ab7f8226f/37b9a3f81453255cc205d89edcdeb79ee6b17d52) and should now be [available for consumption](https://community.chocolatey.org/packages/libreoffice-still/7.5.8). Issue can be closed.

**Comment by zWhdmB5T** [comment]
@brogers5 Thank you for all explanation and help. Yes, `libreoffice-still` works as intended and package is updated and online now. Thank you. And I asked the Document Foundation team if they might set the Update Checker on time for the future. And I got the Update Checker URL working with the appropriate user agent. Thank you!

@brogers5 Sorry that I ping you. Because you have been very helpful in this case, would you mind having an eye on [libreoffice-help](https://community.chocolatey.org/packages/libreoffice-help). Just asking… just dreaming…
